toolkit/tools/internal/isogenerator/isogenerator.go

```diff
@@ -214,32 +216,63 @@ func setUpIsoGrub2Bootloader(info isoGenInfo) (err error) {
 	return nil
 }
 
-func copyShimFromInitrd(info isoGenInfo, efiBootImgTempMountDir, bootBootloaderFile, grubBootloaderFile string,
-) (err error) {
-	bootDirPath := filepath.Join(efiBootImgTempMountDir, "EFI", "BOOT")
+func setUpIsoGrub2Bootloader(info isoGenInfo) (err error) {
 
-	initrdBootBootloaderFilePath := filepath.Join(initrdEFIBootDirectoryPath, bootBootloaderFile)
-	buildDirBootEFIFilePath := filepath.Join(bootDirPath, bootBootloaderFile)
-	err = extractFromInitrdAndCopy(info, initrdBootBootloaderFilePath, buildDirBootEFIFilePath)
+	extractedShimDir, err := os.MkdirTemp(info.Config.BuildDirPath, "extracted-shim")
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to create temporary folder for extracting the shim:\n%w", err)
 	}
+	defer os.RemoveAll(extractedShimDir)
 
-	initrdGrubBootloaderFilePath := filepath.Join(initrdEFIBootDirectoryPath, grubBootloaderFile)
-	buildDirGrubEFIFilePath := filepath.Join(bootDirPath, grubBootloaderFile)
-	err = extractFromInitrdAndCopy(info, initrdGrubBootloaderFilePath, buildDirGrubEFIFilePath)
+	shimFileName := "bootx64.efi"
+	grubFileName := "grubx64.efi"
+
+	if runtime.GOARCH == "arm64" {
```

Comment:
Use a switch statement and error out if an arch isn't supported.

Reply:
Fixed.